hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientJCacheConcurrencyTest.java

```diff
@@ -0,0 +1,131 @@
+
+    private static final int CONCURRENCY_LEVEL = 100;
+
+    private static HazelcastInstance hz1;
+    private static HazelcastInstance hz2;
+    private static HazelcastInstance client;
+
+    private static HazelcastServerCachingProvider cachingProvider1;
+    private static HazelcastServerCachingProvider cachingProvider2;
+    private static HazelcastClientCachingProvider cachingProvider3;
+
+    @BeforeClass
+    public static void init() {
+        Config config = new Config();
+        JoinConfig join = config.getNetworkConfig().getJoin();
+        join.getMulticastConfig().setEnabled(false);
+        join.getTcpIpConfig().setEnabled(true);
+        join.getTcpIpConfig().addMember("127.0.0.1");
+
+        hz1 = Hazelcast.newHazelcastInstance(config);
+        hz2 = Hazelcast.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().addAddress("127.0.0.1");
+        client = HazelcastClient.newHazelcastClient(clientConfig);
+
+        cachingProvider1 = HazelcastServerCachingProvider.createCachingProvider(hz1);
+        cachingProvider2 = HazelcastServerCachingProvider.createCachingProvider(hz2);
+        cachingProvider3 = HazelcastClientCachingProvider.createCachingProvider(client);
+    }
+
+    @AfterClass
+    public static void tear() {
+        cachingProvider1.close();
+        cachingProvider2.close();
+        cachingProvider3.close();
+        HazelcastClient.shutdownAll();
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void test_concurrect_cachemanager_creation() throws Exception {
+        final URI uri = new URI(randomString());
+        final Properties properties = HazelcastCachingProvider.propertiesByInstanceName(client.getName());
+
+        final Semaphore startSemaphore = new Semaphore(CONCURRENCY_LEVEL);
+        final CountDownLatch endLatch = new CountDownLatch(CONCURRENCY_LEVEL);
+        final AtomicReferenceArray<CacheManager> managers = new AtomicReferenceArray<CacheManager>(CONCURRENCY_LEVEL);
+
+        startSemaphore.acquire(100);
+        for (int i = 0; i < CONCURRENCY_LEVEL; i++) {
+            final int index = i;
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Sync all threads
+                        startSemaphore.acquire();
+
+                        // Create CacheManagers
+                        CacheManager cacheManager = cachingProvider3.getCacheManager(uri, null, properties);
+                        managers.set(index, cacheManager);
+
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    } finally {
+                        endLatch.countDown();
+                    }
+                }
+            }).start();
+        }
+
+        startSemaphore.release(100);
+        endLatch.await(10, TimeUnit.MINUTES);
```

Comment:
We have an assertOpenEventually. If you need to wait potentially 10 minutes for this test to complete, this test has is now allowed to be a normal unit test but needs to be moved to the nightly tests.